src/parsers/perf/perfparser.cpp

```diff
@@ -1015,6 +1018,34 @@ class PerfParserPrivate : public QObject
         }
     }
 
+    void addTracepointToTracepointStack(uint32_t tid, const Data::Tracepoint& tracepoint)
+    {
+        auto addToLookup = [this](const QString& name) -> bool {
+            for (const auto& parameters : tracepointParameters) {
+                auto stopName = name;
+                stopName.replace(parameters.startRegex, parameters.stopExpression);
+
+                if (stopName == name) {
+                    continue;
+                }
+
+                tracepointNameRegexCache[name] = stopName;
+                tracepointCostIdLookup[name] = addCostType(parameters.name, Data::Costs::Unit::TracepointCost);
+                return true;
+            }
+            return false;
+        };
+
+        if (!tracepointNameRegexCache.contains(tracepoint.name)) {
```

Comment:
multiple double qhash lookups (once for contains, once for write/lookup) and we should cache negative lookups too. more duplicate lookups in the single place this is being used from (addSample), so rewrite this all as:

```
QString toTracepointStopName(const QString &name)
{
auto cacheIt = tracepointStopNames.find(name);
if (cacheIt == tracepointStopNames.end()) {
    for (const auto& parameters : tracepointParameters) {
        auto stopName = name;
        stopName.replace(parameters.startRegex, parameters.stopExpression);

        if (stopName == name) {
            continue;
        }

        cacheIt = tracepointStopNames.insert(stopName);
        tracepointCostIdLookup[name] = addCostType(parameters.name, Data::Costs::Unit::TracepointCost);
        break;
    }
    if (cacheIt == tracepointStopNames.end()) {
        // not a tracepoint that should be time-measured
        cacheIt = tracepointStopNames.insert({});
    }
}

return *cacheIt;
}
```